r?fzzzy Refactor frame

ianb opened the pull request:
As a whole this fixes #1713

Tried to split up refactoring steps into multiple commits

page/shot/view.js is still pretty hairy, but I moved a couple things out.

I am pretty sure I got everything from #1837 reapplied.  The Share button wraps again though, but will make a followup ticket.

fzzzy commented on server/src/caching.js:
```diff
@@ -7,6 +7,7 @@ if (! config.setCache) {
 }
 
 exports.setCache = function (res, options) {
+  console.log("setcache?", config.setCache, process.env.SET_CACHE);
```
This is probably useful to leave in, but it should be an info. Or take it out if you don't think we always need to see this.

fzzzy commented on server/src/pages/shot/controller.js:
```diff
@@ -1,51 +1,46 @@
       document.dispatchEvent(event);
-      requestProfile();
-      requestHasSavedShot(model.shot.id);
+      //requestHasSavedShot(model.shot.id);
```
I think this should just be removed, to clean up the codebase.

fzzzy replied:
Ok.

fzzzy commented on server/src/pages/shot/controller.js:
```diff
@@ -1,51 +1,46 @@
+      //requestHasSavedShot(model.shot.id);
     }, false);
-    document.addEventListener("refresh-profile", refreshProfile, false);
+    /*
```
Ditto.

fzzzy commented on server/src/pages/shot/controller.js:
```diff
@@ -192,6 +165,7 @@ function sendShowElement(clipId) {
   }
 }
 
+/*
```
Remove.

fzzzy commented on server/src/pages/shot/view.js:
```diff
@@ -350,7 +166,7 @@ class Frame extends React.Component {
   }
 
   onClickUploadFullPage(e) {
```
At this point maybe we should remove more code related to old features that are no longer implemented? The product is firming up a lot and it's unlikely that we will implement new features by using old code, but we will implement them with new code instead.

fzzzy commented on server/src/server.js:
```diff
@@ -641,6 +641,7 @@ app.post("/api/delete-shot", function (req, res) {
   });
 });
 
+/*
```
Let's just remove this too...

fzzzy commented on server/src/server.js:
```diff
@@ -822,31 +824,8 @@ app.get("/contribute.json", function (req, res) {
 });
 
 // FIXME: this can't the right way to do this...
```
Great, the non-right way of doing this is no longer the way we do it! Remove the fixme and the next commented line.

ianb commented:
I addressed the comments except code removal – per my comment above, I'd like to deal with that work separately, as it is a wider scope than the frame.js refactoring.

fzzzy commented on server/src/pages/admin/server.js:
```diff
@@ -7,9 +7,10 @@ let app = express();
+let authorizedAccountIds = ['anon63daec35-fb29-8d49-ac6f-af7763523c0a', 'anon202dc00'];
 
 app.use(function (req, res, next) {
+  req.deviceId = 'anon202dc00'
```
Hmm, I don't think this is right!
